tests/test_batch.py

```diff
@@ -217,5 +219,335 @@ def given_configuration_file_version_different_as_river_file_version_when_runnin
+        assert dfastmi.batch.check_configuration(rivers, config) is False
+        
+    def given_version_with_no_matching_version_when_check_configuration_then_return_false(self):
+        rivers = RiversObject("dfastmi/Dutch_rivers_v1.ini")
+        config = ConfigParser()
         
+        config.add_section("General")
+        config.set("General", "Version", "0.0")
         
+        assert dfastmi.batch.check_configuration(rivers, config) is False
+    
+    class Test_check_configuration_v1():
+        @pytest.fixture
+        def rivers(self):
+            return  RiversObject("dfastmi/Dutch_rivers_v1.ini")
+        
+        @pytest.fixture
+        def config(self):
+            return  ConfigParser()
+                    
+        def given_version_1_when_check_configuration_then_return_false(self, rivers : RiversObject, config : ConfigParser):
+            assert dfastmi.batch.check_configuration(rivers, config) is False
+            
+        def given_general_section_when_check_configuration_then_return_false(self, rivers : RiversObject, config : ConfigParser):
+            self.set_valid_general_section(config, "1.0")
+            
+            assert dfastmi.batch.check_configuration(rivers, config) is False
+            
+        def given_general_section_with_qthreshold_when_check_configuration_then_return_false(self, rivers : RiversObject, config : ConfigParser):
+            self.set_valid_general_section(config, "1.0")
+            config.set("General", "Qthreshold", "100")
+            
+            assert dfastmi.batch.check_configuration(rivers, config) is False
+            
+        def given_general_section_with_qthreshold_and_qbankfull_when_check_configuration_then_return_false(self, rivers : RiversObject, config : ConfigParser):
+            self.set_valid_general_section_with_q_values(config, "1.0")
+            
+            assert dfastmi.batch.check_configuration(rivers, config) is False
+            
+        def given_q_sections_with_discharge_check_configuration_then_return_false(self, rivers : RiversObject, config : ConfigParser):           
+            self.set_valid_general_section_with_q_values(config, "1.0")
+            self.add_q_section(config)
+            
+            assert dfastmi.batch.check_configuration(rivers, config) is False
+            
+        def given_mode_specific_test_with_discharge_check_configuration_then_return_true(self, rivers : RiversObject, config : ConfigParser):
+            self.set_valid_general_section_with_q_values(config, "1.0")
+            self.add_q_section(config)
+            config.set("General", "mode", "test")
+            
+            assert dfastmi.batch.check_configuration(rivers, config) is True
+            
+        def set_valid_general_section(self, config : ConfigParser, version : str):
+            config.add_section("General")
+            config.set("General", "Version", version)
+            config.set("General", "Branch", "Bovenrijn & Waal")
```

Comment:
do not use real key value, use fake if you make test value your self. I believe using real values should only be used in acceptance test to users. just make up your own name. Use team members as example, or you favorite kids tv show

Reply:
Because this is a data acces test and very much dependent on the Dutch_rivers_v1 which has this data in it

Reply:
I can make my own file of  Dutch_rivers_v1.ini, but I do not think that is worth the effort